src/tic-tac-toe.ts

```diff
@@ -40,36 +44,98 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
+    for (const rowIndex in this.board) {
+      if (this.board[rowIndex].every(cell => this.isCurrentPlayer(cell))) {
+        winData.hasWinner = true;
+        for (
+          let columnIndex = 0;
+          columnIndex < this.board.length;
+          columnIndex++
+        ) {
+          winData.winningCells.push([Number(rowIndex), columnIndex]);
+        }
+        break;
+      }
+    }
+
+    return winData;
   }
 
-  private checkColumnsForWin(): boolean {
+  private checkColumnsForWin(): WinData {
+    const winData: WinData = {
+      hasWinner: false,
+      winningCells: [],
+    };
+
     const transposedBoard = this.eachIndex().map(i =>
       this.board.map(row => row[i])
     );
 
-    return transposedBoard.some(row => row.every(this.isCurrentPlayer));
+    winData.hasWinner = transposedBoard.some(row =>
+      row.every(this.isCurrentPlayer)
+    );
+
+    if (winData.hasWinner) {
+      transposedBoard.forEach((row, rowIndex) => {
+        if (row.every(value => this.isCurrentPlayer(value))) {
+          const vectors = [];
+          for (let i = 0; i < row.length; i++) {
+            vectors.push([rowIndex, i]);
+          }
+          winData.winningCells = transposeVectors(vectors);
+        }
+      });
+    }
+
+    return winData;
   }
 
-  private checkDiagonalsForWin(): boolean {
-    const diagonals = [
-      this.eachIndex().map(i => this.board[i][i]),
-      this.eachIndex().map(i => this.board[i][this.boardSize - 1 - i]),
-    ];
+  private checkDiagonalsForWin(): WinData {
+    const leftToRightDiagonalWin = this.eachIndex()
+      .map(i => this.board[i][i])
+      .every(this.isCurrentPlayer);
+
+    const rightToLeftDiagonalWin = this.eachIndex()
+      .map(i => this.board[i][this.boardSize - 1 - i])
+      .every(this.isCurrentPlayer);
+
+    const winData: WinData = {
+      hasWinner: leftToRightDiagonalWin || rightToLeftDiagonalWin,
+      winningCells: [],
+    };
+
+    if (winData.hasWinner) {
+      if (leftToRightDiagonalWin) {
+        winData.winningCells = this.eachIndex().map(i => [i, i]);
+      } else {
+        winData.winningCells = this.eachIndex().map(i => [
+          i,
+          this.boardSize - 1 - i,
+        ]);
+      }
+    }
 
-    return diagonals.some(values => values.every(this.isCurrentPlayer));
+    return winData;
   }
 
   private checkGameForWin(): void {
-    const winCondition = [
+    const winChecks = [
       this.checkRowsForWin(),
       this.checkColumnsForWin(),
-      this.checkDiagonalsForWin()
+      this.checkDiagonalsForWin(),
     ];
 
-    if (winCondition.some(winnerCheck => winnerCheck)) {
-      this.winner = this.currentPlayer;
+    for (const check of winChecks) {
```

Comment:
https://www.geeksforgeeks.org/javascript-array-prototype-find-function/ would do it, no?